Intial configuration

msotoperdomo opened the pull request:
Initial configuration for the base image for php applications.
- Include Consul template 
- Include base php libraries
- Php.ini and php-fpm.conf defined as consul keys
- Base libraries use by applications
- Documentation

Commit:
Initial configuration for the image deployment

Commit:
Fix test image building

Commit:
Building optimization

Commit:
README documentation

nhuray commented on bin/functions:
```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env bash
```
This file shouldn't be commited ! You have to add it to the `.gitignore`

nhuray commented on rootfs/usr/local/bin/reload-config.sh:
```diff
@@ -0,0 +1,9 @@
+pid_php=$(cat /usr/local/var/run/php-fpm.pid)
+if [ -n "$pid_php" ]; then
+  echo "Reloading php configuration"
+  kill -USR2 $pid_php
```
Cool stuff !

nhuray commented on versions/1.0.0/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+      libmcrypt-dev \
+      libpng-dev \
+      curl wget bash tree jq bind-tools su-exec \
+    && apk add --no-cache --update libmemcached-libs zlib \
```
Why do you need to run those commands ? They are not defined in your base image ?

nhuray commented on bin/docker-php:
```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env bash
```
For the other Docker images, we named the script based on the name of the image, so `php` instead of `docker-php` !

nhuray reviewed (changes requested):
Some changes ! Great job 👍

bratislavml commented on rootfs/usr/local/bin/reload-config.sh:
```diff
@@ -0,0 +1,9 @@
+echo "Reload script"
+pid_php=$(cat /usr/local/var/run/php-fpm.pid)
+if [ -n "$pid_php" ]; then
+  echo "Reloading php configuration"
```
👍

bratislavml reviewed (approved):
👍 OK nothing to add beside @nhuray

Commit:
Use of php instead of docker-php as image name

Commit:
- Added a sleep so the template it's present the first time the service start and the reload configuration takes the new configuration
- Rename the php-fpm references to match the image name definition
- Change the reference from the alpine image from latest to 3.6
